# nf-core Strict Syntax Health Report

This repository tracks the health of nf-core pipelines with respect to Nextflow's strict syntax linting.

**Last updated:** 2026-01-12 16:52:51 UTC

**Total:** 4 parse errors, 4146 errors, 9669 warnings across 129 pipelines

## Trends

### Errors

![Errors Chart](errors_chart.png)

### Warnings

![Warnings Chart](warnings_chart.png)

## Results

| Pipeline                                                                          | Parse Error | Errors | Warnings |
| --------------------------------------------------------------------------------- | :---------: | -----: | -------: |
| [proteinannotator](https://github.com/nf-core/proteinannotator)                   |     No      |      1 |        9 |
| [createpanelrefs](https://github.com/nf-core/createpanelrefs)                     |     No      |      1 |       15 |
| [rnavar](https://github.com/nf-core/rnavar)                                       |     No      |      1 |       17 |
| [ribomsqc](https://github.com/nf-core/ribomsqc)                                   |     No      |      1 |       27 |
| [createtaxdb](https://github.com/nf-core/createtaxdb)                             |     No      |      1 |       35 |
| [mag](https://github.com/nf-core/mag)                                             |     No      |      1 |       35 |
| [molkart](https://github.com/nf-core/molkart)                                     |     No      |      1 |       63 |
| [methylong](https://github.com/nf-core/methylong)                                 |     No      |      1 |       81 |
| [seqinspector](https://github.com/nf-core/seqinspector)                           |     No      |      2 |       14 |
| [seqsubmit](https://github.com/nf-core/seqsubmit)                                 |     No      |      2 |       32 |
| [taxprofiler](https://github.com/nf-core/taxprofiler)                             |     No      |      2 |      140 |
| [scdownstream](https://github.com/nf-core/scdownstream)                           |     No      |      3 |       30 |
| [reportho](https://github.com/nf-core/reportho)                                   |     No      |      3 |       90 |
| [methylarray](https://github.com/nf-core/methylarray)                             |     No      |      4 |       19 |
| [methylseq](https://github.com/nf-core/methylseq)                                 |     No      |      4 |       65 |
| [gwas](https://github.com/nf-core/gwas)                                           |     No      |      6 |       16 |
| [kmermaid](https://github.com/nf-core/kmermaid)                                   |     No      |      6 |       16 |
| [mitodetect](https://github.com/nf-core/mitodetect)                               |     No      |      6 |       16 |
| [panoramaseq](https://github.com/nf-core/panoramaseq)                             |     No      |      6 |       16 |
| [troughgraph](https://github.com/nf-core/troughgraph)                             |     No      |      6 |       16 |
| [variantprioritization](https://github.com/nf-core/variantprioritization)         |     No      |      7 |       14 |
| [demo](https://github.com/nf-core/demo)                                           |     No      |      7 |       17 |
| [fastquorum](https://github.com/nf-core/fastquorum)                               |     No      |      7 |       28 |
| [bamtofastq](https://github.com/nf-core/bamtofastq)                               |     No      |      7 |       42 |
| [lsmquant](https://github.com/nf-core/lsmquant)                                   |     No      |      7 |       54 |
| [tfactivity](https://github.com/nf-core/tfactivity)                               |     No      |      7 |       57 |
| [longraredisease](https://github.com/nf-core/longraredisease)                     |     No      |      7 |      121 |
| [phaseimpute](https://github.com/nf-core/phaseimpute)                             |     No      |      7 |      237 |
| [cellpainting](https://github.com/nf-core/cellpainting)                           |     No      |      8 |       35 |
| [pairgenomealign](https://github.com/nf-core/pairgenomealign)                     |     No      |      9 |       36 |
| [tumourevo](https://github.com/nf-core/tumourevo)                                 |     No      |     10 |       56 |
| [mhcquant](https://github.com/nf-core/mhcquant)                                   |     No      |     10 |       80 |
| [proteinfamilies](https://github.com/nf-core/proteinfamilies)                     |     No      |     11 |       19 |
| [nanostring](https://github.com/nf-core/nanostring)                               |     No      |     11 |       35 |
| [rangeland](https://github.com/nf-core/rangeland)                                 |     No      |     11 |       79 |
| [ampliseq](https://github.com/nf-core/ampliseq)                                   |     No      |     11 |      246 |
| [smrnaseq](https://github.com/nf-core/smrnaseq)                                   |     No      |     12 |       74 |
| [pacvar](https://github.com/nf-core/pacvar)                                       |     No      |     13 |       56 |
| [drop](https://github.com/nf-core/drop)                                           |     No      |     13 |       77 |
| [variantbenchmarking](https://github.com/nf-core/variantbenchmarking)             |     No      |     13 |      144 |
| [sopa](https://github.com/nf-core/sopa)                                           |     No      |     14 |       17 |
| [drugresponseeval](https://github.com/nf-core/drugresponseeval)                   |     No      |     14 |       41 |
| [tbanalyzer](https://github.com/nf-core/tbanalyzer)                               |     No      |     14 |       47 |
| [detaxizer](https://github.com/nf-core/detaxizer)                                 |     No      |     14 |       61 |
| [pangenome](https://github.com/nf-core/pangenome)                                 |     No      |     14 |       63 |
| [rnafusion](https://github.com/nf-core/rnafusion)                                 |     No      |     14 |      116 |
| [funcscan](https://github.com/nf-core/funcscan)                                   |     No      |     14 |      124 |
| [circrna](https://github.com/nf-core/circrna)                                     |     No      |     14 |      157 |
| [clipseq](https://github.com/nf-core/clipseq)                                     |     No      |     15 |        0 |
| [dualrnaseq](https://github.com/nf-core/dualrnaseq)                               |     No      |     15 |        0 |
| [imcyto](https://github.com/nf-core/imcyto)                                       |     No      |     15 |        0 |
| [mnaseseq](https://github.com/nf-core/mnaseseq)                                   |     No      |     15 |        0 |
| [proteogenomicsdb](https://github.com/nf-core/proteogenomicsdb)                   |     No      |     15 |        0 |
| [spatialvi](https://github.com/nf-core/spatialvi)                                 |     No      |     15 |       20 |
| [hlatyping](https://github.com/nf-core/hlatyping)                                 |     No      |     15 |       23 |
| [diaproteomics](https://github.com/nf-core/diaproteomics)                         |     No      |     16 |        0 |
| [cageseq](https://github.com/nf-core/cageseq)                                     |     No      |     17 |        0 |
| [slamseq](https://github.com/nf-core/slamseq)                                     |     No      |     17 |        0 |
| [genomeassembler](https://github.com/nf-core/genomeassembler)                     |     No      |     17 |       93 |
| [phyloplace](https://github.com/nf-core/phyloplace)                               |     No      |     18 |      106 |
| [scrnaseq](https://github.com/nf-core/scrnaseq)                                   |     No      |     18 |      126 |
| [datasync](https://github.com/nf-core/datasync)                                   |     No      |     19 |       20 |
| [metapep](https://github.com/nf-core/metapep)                                     |     No      |     20 |       38 |
| [epitopeprediction](https://github.com/nf-core/epitopeprediction)                 |     No      |     21 |       36 |
| [mcmicro](https://github.com/nf-core/mcmicro)                                     |     No      |     22 |       48 |
| [sammyseq](https://github.com/nf-core/sammyseq)                                   |     No      |     22 |      148 |
| [fastqrepair](https://github.com/nf-core/fastqrepair)                             |     No      |     26 |       28 |
| [abotyper](https://github.com/nf-core/abotyper)                                   |     No      |     27 |       64 |
| [magmap](https://github.com/nf-core/magmap)                                       |     No      |     27 |       70 |
| [bactmap](https://github.com/nf-core/bactmap)                                     |     No      |     28 |       64 |
| [bacass](https://github.com/nf-core/bacass)                                       |     No      |     28 |      131 |
| [stableexpression](https://github.com/nf-core/stableexpression)                   |     No      |     29 |       36 |
| [viralmetagenome](https://github.com/nf-core/viralmetagenome)                     |     No      |     30 |      246 |
| [denovotranscript](https://github.com/nf-core/denovotranscript)                   |     No      |     31 |       51 |
| [alleleexpression](https://github.com/nf-core/alleleexpression)                   |     No      |     32 |       37 |
| [multiplesequencealign](https://github.com/nf-core/multiplesequencealign)         |     No      |     32 |      158 |
| [rnadnavar](https://github.com/nf-core/rnadnavar)                                 |     No      |     32 |      371 |
| [rarevariantburden](https://github.com/nf-core/rarevariantburden)                 |     No      |     33 |       22 |
| [nascent](https://github.com/nf-core/nascent)                                     |     No      |     33 |      161 |
| [raredisease](https://github.com/nf-core/raredisease)                             |     No      |     33 |      373 |
| [genomeqc](https://github.com/nf-core/genomeqc)                                   |     No      |     34 |      101 |
| [proteomicslfq](https://github.com/nf-core/proteomicslfq)                         |     No      |     36 |        0 |
| [isoseq](https://github.com/nf-core/isoseq)                                       |     No      |     36 |       13 |
| [references](https://github.com/nf-core/references)                               |     No      |     36 |       57 |
| [circdna](https://github.com/nf-core/circdna)                                     |     No      |     38 |       30 |
| [coproid](https://github.com/nf-core/coproid)                                     |     No      |     38 |       58 |
| [genomeskim](https://github.com/nf-core/genomeskim)                               |     No      |     40 |       13 |
| [lncpipe](https://github.com/nf-core/lncpipe)                                     |     No      |     41 |      176 |
| [omicsgenetraitassociation](https://github.com/nf-core/omicsgenetraitassociation) |     No      |     44 |       30 |
| [viralrecon](https://github.com/nf-core/viralrecon)                               |     No      |     44 |      188 |
| [pathogensurveillance](https://github.com/nf-core/pathogensurveillance)           |     No      |     44 |      486 |
| [radseq](https://github.com/nf-core/radseq)                                       |     No      |     45 |       42 |
| [meerpipe](https://github.com/nf-core/meerpipe)                                   |     No      |     45 |       84 |
| [crisprseq](https://github.com/nf-core/crisprseq)                                 |     No      |     46 |       92 |
| [fetchngs](https://github.com/nf-core/fetchngs)                                   |     No      |     47 |       25 |
| [differentialabundance](https://github.com/nf-core/differentialabundance)         |     No      |     48 |       10 |
| [spinningjenny](https://github.com/nf-core/spinningjenny)                         |     No      |     50 |        9 |
| [riboseq](https://github.com/nf-core/riboseq)                                     |     No      |     53 |      158 |
| [readsimulator](https://github.com/nf-core/readsimulator)                         |     No      |     54 |       49 |
| [demultiplex](https://github.com/nf-core/demultiplex)                             |     No      |     55 |       42 |
| [variantcatalogue](https://github.com/nf-core/variantcatalogue)                   |     No      |     55 |       48 |
| [hgtseq](https://github.com/nf-core/hgtseq)                                       |     No      |     58 |       58 |
| [nanoseq](https://github.com/nf-core/nanoseq)                                     |     No      |     59 |       60 |
| [phageannotator](https://github.com/nf-core/phageannotator)                       |     No      |     60 |       98 |
| [proteinfold](https://github.com/nf-core/proteinfold)                             |     No      |     63 |       71 |
| [metatdenovo](https://github.com/nf-core/metatdenovo)                             |     No      |     65 |      157 |
| [spatialxe](https://github.com/nf-core/spatialxe)                                 |     No      |     66 |       25 |
| [metaboigniter](https://github.com/nf-core/metaboigniter)                         |     No      |     66 |      131 |
| [diseasemodulediscovery](https://github.com/nf-core/diseasemodulediscovery)       |     No      |     70 |       80 |
| [hic](https://github.com/nf-core/hic)                                             |     No      |     74 |       65 |
| [scnanoseq](https://github.com/nf-core/scnanoseq)                                 |     No      |     74 |      140 |
| [evexplorer](https://github.com/nf-core/evexplorer)                               |     No      |     75 |       53 |
| [rnasplice](https://github.com/nf-core/rnasplice)                                 |     No      |     77 |      175 |
| [chipseq](https://github.com/nf-core/chipseq)                                     |     No      |     79 |      178 |
| [viralintegration](https://github.com/nf-core/viralintegration)                   |     No      |     80 |       24 |
| [marsseq](https://github.com/nf-core/marsseq)                                     |     No      |     81 |       61 |
| [callingcards](https://github.com/nf-core/callingcards)                           |     No      |     81 |      163 |
| [genomeannotator](https://github.com/nf-core/genomeannotator)                     |     No      |     85 |      148 |
| [rnaseq](https://github.com/nf-core/rnaseq)                                       |     No      |     85 |      299 |
| [hicar](https://github.com/nf-core/hicar)                                         |     No      |     89 |      122 |
| [eager](https://github.com/nf-core/eager)                                         |     No      |     96 |        0 |
| [cutandrun](https://github.com/nf-core/cutandrun)                                 |     No      |     97 |       75 |
| [atacseq](https://github.com/nf-core/atacseq)                                     |     No      |    118 |      200 |
| [deepmutscan](https://github.com/nf-core/deepmutscan)                             |     No      |    122 |      118 |
| [oncoanalyser](https://github.com/nf-core/oncoanalyser)                           |     No      |    306 |      102 |
| [airrflow](https://github.com/nf-core/airrflow)                                   |     Yes     |      - |        - |
| [deepmodeloptim](https://github.com/nf-core/deepmodeloptim)                       |     Yes     |      - |        - |
| [pixelator](https://github.com/nf-core/pixelator)                                 |     Yes     |      - |        - |
| [sarek](https://github.com/nf-core/sarek)                                         |     Yes     |      - |        - |

## About

This report is generated weekly by running `nextflow lint` on each nf-core pipeline.
The linting checks for strict syntax compliance in Nextflow DSL2 code.

- **Parse errors** indicate pipelines where `nextflow lint` could not run at all, typically due to syntax errors that prevent Nextflow from parsing the pipeline code
- **Errors** indicate syntax issues that will cause problems in future Nextflow versions
- **Warnings** indicate deprecated patterns that should be updated
